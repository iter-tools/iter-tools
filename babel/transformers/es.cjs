const { resolve } = require('path');
const babelJest = require('babel-jest');

module.exports = babelJest.createTransformer({
  configFile: resolve(__dirname, '../jest.config.cjs'),
});
