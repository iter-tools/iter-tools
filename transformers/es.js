const createTransformer = require('./create-transformer');

module.exports = createTransformer('babel-es.config.js');
