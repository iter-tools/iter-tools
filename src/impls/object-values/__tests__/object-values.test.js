import { objectValues } from '@iter-tools/es';

describe('objectValues', () => {
  it('works with Objects', () => {
    const i = objectValues({ 1: 1, 2: 2 });
    expect(Array.from(i)).toEqual([1, 2]);
  });

  it('can be reused', () => {
    const i = objectValues({ 1: 1, 2: 2 });
    expect(Array.from(i)).toEqual([1, 2]);
    expect(Array.from(i)).toEqual([1, 2]);
  });

  it('works with null', () => {
    const i = objectValues(null);
    expect(Array.from(i)).toEqual([]);
  });

  it('throws with non-object', () => {
    const obj: any = 4;
    expect(() => objectValues(obj)).toThrowErrorMatchingSnapshot();
  });
});
