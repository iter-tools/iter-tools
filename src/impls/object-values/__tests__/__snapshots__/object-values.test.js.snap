// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objectValues throws with non-object 1`] = `"obj argument to objectValues must be an object, null, or undefined"`;
